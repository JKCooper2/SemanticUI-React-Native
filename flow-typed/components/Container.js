// @flow
import type { Node } from 'react';
import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import type { TextAlignType } from '../styles';

export type ContainerPropsType = {
    children?: Node,
    content?: Node,
    fluid: boolean,
    text: boolean,
    textAlign: TextAlignType,
    style?: StyleObj
  };