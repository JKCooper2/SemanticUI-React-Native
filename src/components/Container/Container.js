// @flow
import React, { PureComponent } from 'react';

import { View } from 'react-native';

import { childrenUtils } from '~/libs';
import { spacing } from '~/styles';

import type { ContainerPropsType } from '~/types/components/Container';

class Container extends PureComponent<ContainerPropsType> {
    static defaultProps = {
        fluid: false,
        text: false,
        textAlign: 'left',
    };

    render() {
        const { children, content, fluid, text, textAlign, style } = this.props;

        const padding = text && { paddingLeft: spacing.padding.large, paddingRight: spacing.padding.large };

        return (
            <View style={[{ flex: fluid ? 1 : 0 }, padding, spacing.getAlignment(textAlign), style]}>
                {childrenUtils.isNil(children) ? content : children}
            </View>
        );
    }
}

export default Container;