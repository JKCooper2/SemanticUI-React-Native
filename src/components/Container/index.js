import Container from './Container.js';

export {
    Container as default
};