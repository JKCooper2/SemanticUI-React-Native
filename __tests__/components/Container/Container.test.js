// @flow
import React from 'react';
import { shallow } from 'enzyme';

import Container from '~/components/Container';

test('renders correctly', () => {
    const container = shallow(<Container />);
    expect(container).toMatchSnapshot();
});
