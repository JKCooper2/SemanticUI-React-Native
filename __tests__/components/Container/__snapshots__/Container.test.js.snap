// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 0,
      },
      Object {
        "alignSelf": "flex-start",
      },
      false,
      undefined,
    ]
  }
/>
`;
